components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.html

```diff
@@ -0,0 +1,13 @@
+<svg #svg [attr.viewBox]="viewBox" 
+    [ngStyle]="{'width': width}" 
+    (window:resize)="onResize()">
```

Comment:
This reminds me of a potential future issue with the compliance line chart as well. If the parent container of the chart should change its width for whatever reason (maybe an extra panel opens or something) then the chart will not resize itself because the overall viewport is not being resized:

![](https://user-images.githubusercontent.com/479121/81618496-b6433980-93a4-11ea-9da2-fd2ee75c1922.gif)

However with the current dashboard layout and how it resizes its layout when opening side panels, the parent container of this chart still remains the same size so relying on `window:resize` is okay for now just like it's currently okay in the compliance reports view. But it could perhaps be an issue in the future should things change with the design so wanted to make a note of this elsewhere besides my own brain. It'll be trickier to implement but I know there are ways to avoid needing to manually redraw based on viewport or parent container width, giving us more flexible and performant charts to work with.

Reply:
This is a really good snag.  Thanks for documenting.  I'll leave for now because of the reasons you mention, but will keep in mind.  I'll do a little research on this when I have the opportunity.

Reply:
oh interesting! 👍

Reply:
[ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) could offer a solution if a polyfill exists for IE.